client/client.go

```diff
@@ -1069,17 +1069,23 @@ func (c *Client) updateNodeStatus() error {
 
 // updateAllocStatus is used to update the status of an allocation
 func (c *Client) updateAllocStatus(alloc *structs.Allocation) {
-	// Only send the fields that are updatable by the client.
+	select {
+	case c.allocUpdates <- alloc:
+	case <-c.shutdownCh:
+	}
+}
+
+// stripAllocation strips the allocation of fields which can be re-assembled in
+// the server.
+func (c *Client) stripAllocation(alloc *structs.Allocation) *structs.Allocation {
```

Comment:
Also the `updateAllocStatus` pushes the alloc to a chan, and we are stripping the allocations when we receive them from the chan so it should be fine.